firewood/src/db.rs

```diff
@@ -510,47 +510,32 @@ impl Db {
         let merkle_path = file::touch_dir("merkle", &db_path)?;
         let merkle_meta_path = file::touch_dir("meta", &merkle_path)?;
         let merkle_payload_path = file::touch_dir("compact", &merkle_path)?;
-
         let root_hash_path = file::touch_dir("root_hash", &db_path)?;
 
-        let file0 = crate::file::File::new(0, SPACE_RESERVED, &merkle_meta_path)?;
-        let fd0 = file0.as_fd();
+        let meta_file = crate::file::File::new(0, SPACE_RESERVED, &merkle_meta_path)?;
+        let meta_fd = meta_file.as_fd();
 
         if reset {
-            // initialize dbparams
+            // initialize DbParams
             if cfg.payload_file_nbit < cfg.payload_regn_nbit
                 || cfg.payload_regn_nbit < PAGE_SIZE_NBIT
             {
                 return Err(DbError::InvalidParams);
             }
-            Self::initialize_header_on_disk(&cfg, fd0)?;
+            Self::initialize_header_on_disk(&cfg, meta_fd)?;
         }
 
         // read DbParams
         let mut header_bytes = [0; size_of::<DbParams>()];
-        nix::sys::uio::pread(fd0, &mut header_bytes, 0).map_err(DbError::System)?;
-        drop(file0);
+        nix::sys::uio::pread(meta_fd, &mut header_bytes, 0).map_err(DbError::System)?;
+        drop(meta_file);
         #[allow(clippy::indexing_slicing)]
         let params: DbParams = cast_slice(&header_bytes)[0];
 
-        let wal = WalConfig::builder()
-            .file_nbit(params.wal_file_nbit)
-            .block_nbit(params.wal_block_nbit)
-            .max_revisions(cfg.wal.max_revisions)
-            .build();
```

Comment:
This is actually a wal config, not a wal. Renamed and moved below, closer to usage